src/fastga_he/gui/power_train_network_viewer.py

```diff
@@ -1,54 +1,184 @@
 # This file is part of FAST-OAD_CS23-HE : A framework for rapid Overall Aircraft Design of Hybrid
 # Electric Aircraft.
-# Copyright (C) 2022 ISAE-SUPAERO
+# Copyright (C) 2025 ISAE-SUPAERO
 
 import os
 import os.path as pth
 
-import networkx as nx
-from pyvis.network import Network
+from pathlib import Path
+import re
+import pygraphviz as pgv
+from bokeh.plotting import figure, output_file, save
+from bokeh.models import (
+    ColumnDataSource,
+    HoverTool,
+    BoxSelectTool,
+    LabelSet,
+)
 
 from fastga_he.powertrain_builder.powertrain import FASTGAHEPowerTrainConfigurator
 
 from . import icons
 
+BACKGROUND_COLOR_CODE = "#bebebe"
+ELECTRICITY_CURRENT_COLOR_CODE = "#007BFF"
+FUEL_FLOW_COLOR_CODE = "#FF5722"
+MECHANICAL_POWER_COLOR_CODE = "#2E7D32"
+
 # Image URLs for graph nodes
 
 icons_dict = {
```

Comment:
Given that we now have more than one attribute per element, wouldn't it be better to switch to a list ?